R/utils.R

```diff
@@ -117,6 +117,14 @@ do_patterns = function(pat_sub, all_cols) {
   return(matched)
 }
 
+# check UTC status
+is_utc = function(tz) {
+  # via grep('UTC|GMT', OlsonNames(), value = TRUE)
+  utc_tz = c("Etc/GMT", "Etc/UTC", "GMT", "GMT-0", "GMT+0", "GMT0", "UTC")
```

Comment:
I sorted this alphabetically... perhaps we should sort it by expected frequency instead? Does `%chin%` have a short-out to return once the first match is found?

Reply:
No. Good idea. If its LHS is length 1, then it should just loop through the RHS and stop early if and when it is found.  (Btw, sorting the input was good thought but doesn't help as it's an order-n two-pass approach using the now-well- established truelength clobber technique.)

Reply:
When `%chin%` does short-circuit as you suggested, then yes this input should be sorted by expected frequency.

Reply:
Done: #4121